### PR TITLE
kie-issues#1448: `matches()` function wrongly behaves

### LINKED ISSUE #1448: `matches()` function wrongly behaves
According to the DMN specs, the `matches()` and `replace()` functions should behave according to the xQuery 1.0 specification. 
Our current implementation doesn't rely on that, using the native Java implementation to manage both.
Unfortunately, the Java regex management differs from the xQuery specs, and natively implementing the xQuery specs leads to huge effort (and complex) work. For that reason, we agreed to rely on an external library that already implemented xQuery specs in Java. 
After a quick analysis (https://en.wikipedia.org/wiki/XQuery#Implementations), we choose [Saxon-He](https://github.com/Saxonica/Saxon-HE) as the best fit for this purpose. 

The scope of the ticket is to integrate this new external dependency into our code base, so both `matches()` and `replace()` can rely on that implementation to correctly behave.

=== _Original description_  ====

TCK Tests revealed some cases where matches() wrongly behaves.

1.     matches("O", "[A-Z-[OI]]", "i") should return false. Before fixing that, double-check in the specs that this is the expected behavior
2.     matches("i", "[A-Z-[OI]]", "i") should return false. Before fixing that, double-check in the specs that this is the expected behavior
 
 
This syntax is invalid for java Pattern
`[A-Z-[OI]]`
and correct one is
`[A-Z&&[^OI]]` (see **_Character classes_** in [docs](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html))

[XML Schema Part 2: Datatypes Second Edition](https://www.w3.org/TR/xmlschema-2/)
makes a distinction between _negation_ (`ˆ`) and _subtraction_ (`-`)
See "Negative Character Group" and "Character Class Subtraction" at [regex chapter](https://www.w3.org/TR/xmlschema-2/#regexs).

_**A negative character group is a [·positive character group·](https://www.w3.org/TR/xmlschema-2/#dt-poschargroup) preceded by the ^ character. For all [·positive character group·](https://www.w3.org/TR/xmlschema-2/#dt-poschargroup)s P, ^P is a valid negative character group, and C(^P) contains all XML characters that are not in C(P).**_

_**A character class subtraction is a [·character class expression·](https://www.w3.org/TR/xmlschema-2/#dt-charexpr) subtracted from a [·positive character group·](https://www.w3.org/TR/xmlschema-2/#dt-poschargroup) or [·negative character group·](https://www.w3.org/TR/xmlschema-2/#dt-negchargroup), using the - character.**_

A "translation" between the different syntax is required

### PR DESCRIPTION
**Pull request by yesamer**
Closes: https://github.com/apache/incubator-kie-issues/issues/1448

According to the DMN specs, the matches() and replace() functions should behave according to the xQuery 1.0 specification.
Our current implementation doesn't rely on that, using the native Java implementation to manage both.
Unfortunately, the Java regex management differs from the xQuery specs, and natively implementing the xQuery specs leads to huge effort (and complex) work. For that reason, we agreed to rely on an external library that already implemented xQuery specs in Java.
After a quick analysis (https://en.wikipedia.org/wiki/XQuery#Implementations), we choose [Saxon-He](https://github.com/Saxonica/Saxon-HE) as the best fit for this purpose.

The scope of the ticket is to integrate this new external dependency into our code base, so that both matches() and replace() can rely on that implementation to correctly behave.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class XQueryImplUtilTest {
+
+    @Test
+    void executeMatchesFunctionTest() {
+        Boolean retrieved = XQueryImplUtil.executeMatchesFunction("test", "^test", "i");
```
Comment (gitgabrio) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java]: Please take notice of [this](https://github.com/apache/incubator-kie-drools/pull/6083#pullrequestreview-2306164537) comment

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                .isInstanceOf(IllegalArgumentException.class).hasMessageContaining("Unrecognized flag");
+
+        assertThatThrownBy(() -> XQueryImplUtil.executeReplaceFunction("test", "(?=\\s)", "ttt", null))
+                .isInstanceOf(IllegalArgumentException.class).hasMessageContaining("No expression before quantifier");
+
+        assertThatThrownBy(() -> XQueryImplUtil.executeReplaceFunction("test", "(.)\\2", "ttt", null))
+                .isInstanceOf(IllegalArgumentException.class).hasMessageContaining("invalid backreference \\2");
+    }
+
+}
```
Comment (gitgabrio) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java:117]: Could you please introduce a test for XQueryImplUtil.evaluateXQueryExpression, to verify that expected exception is thrown when Class<T> expectedTypeResult is neither Boolean nor String ?
Thanks!

**Review by gitgabrio — CHANGES_REQUESTED** [review]
Thanks @yesamer 
I would ask just a couple minor adjustments

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java (hunk 2 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java:117]: Good point

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-drools -u https://github.com/apache/incubator-kie-drools/pull/6085 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-6085/1/display/redirect

**Test results:**
- PASSED: 22180
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-6085/1/testReport/org.kie.maven.plugin.it/kie-maven-plugin-test-kjar-2/kie_maven_plugin_test_kjar_2/">org.kie.maven.plugin.it.kie-maven-plugin-test-kjar-2.kie-maven-plugin-test-kjar-2</a></summary>
The build exited with code 143. See /home/jenkins/jenkins-agent/workspace/llrequest_jobs_drools-pr_PR-6085/kogito-pipelines/bc/apache_incubator-kie-drools/kie-maven-plugin/target/it/kie-maven-plugin-test-kjar-2/build.log for details.
</details>

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java (hunk 1 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java]: @gitgabrio Done

**Comment by yesamer** [comment]
@gitgabrio Tests fixed.

**Review by gitgabrio — APPROVED** [review]
Thx @yesamer !

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java [hunk 3]
```diff
@@ -19,116 +19,146 @@
...
-    @Test
-    void invokeNull() {
-        assertThatExceptionOfType(InvalidParameterException.class).isThrownBy(() -> MatchesFunction.matchFunctionWithFlags(null, null, null));
-        assertThatExceptionOfType(InvalidParameterException.class).isThrownBy(() -> MatchesFunction.matchFunctionWithFlags(null, "test", null));
-        assertThatExceptionOfType(InvalidParameterException.class).isThrownBy(() -> MatchesFunction.matchFunctionWithFlags("test", null, null));
+    private static final MatchesFunction matchesFunction = MatchesFunction.INSTANCE;
+
+    @ParameterizedTest
+    @MethodSource("invokeNullTestData")
+    void invokeNullTest(String input, String pattern, Class<?> expectedErrorEventClass) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: Inline parameter expectedErrorEventClass. it is always going to be InvalidParametersEvent.class.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java [hunk 4]
```diff
@@ -19,116 +19,146 @@
...
-    @Test
-    void invokeWithoutFlagsMatch() {
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("test", "test",null), true);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("foobar", "^fo*b",null), true);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("abracadabra", "bra", ""), true);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("abracadabra", "bra",null), true);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("(?xi)[hello world()]", "hello",null), true);
+    @ParameterizedTest
+    @MethodSource("invokeUnsupportedFlagsTestData")
+    void invokeUnsupportedFlagsTest(String input, String pattern, String flags, Class<?> expectedErrorEventClass) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: Inline parameter expectedErrorEventClass. it is always going to be InvalidParametersEvent.class.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java [hunk 5]
```diff
@@ -19,116 +19,146 @@
...
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("fo\nbar", "fo.bar",null), false);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("h", "(.)\3",null), false);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("h", "(.)\2",null), false);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("input", "\3",null), false);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("fo\nbar", "(?iU)(?iU)(ab)[|cd]",null), false);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("fo\nbar", "(?x)(?i)hello world","i"), false);
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("fo\nbar", "(?xi)hello world",null), false);
+    @ParameterizedTest
+    @MethodSource("invokeWithoutFlagsMatchTestData")
+    void invokeWithoutFlagsMatchTest(String input, String pattern, Boolean expectedResult) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java:63]: Inline parameter expected result. it is always going to be true.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java [hunk 6]
```diff
@@ -19,116 +19,146 @@
...
+                { "(?xi)[hello world()]", "hello", true }
+        };
     }
 
-    @Test
-    void invokeWithFlagMultiline() {
-        FunctionTestUtil.assertResult(MatchesFunction.matchFunctionWithFlags("fo\nbar", "^bar", "m"), true);
+    @ParameterizedTest
+    @MethodSource("invokeNullOrEmptyFlagsMatchTestData")
+    void invokeNullOrEmptyFlagsMatchTest(String input, String pattern, String flags, Boolean expectedResult) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java:79]: Inline parameter expected result. it is always going to be true.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java [hunk 7]
```diff
@@ -19,116 +19,146 @@
...
+                { "foobar", "^fo*b", null, true },
+                { "abracadabra", "bra", null, true },
+                { "abracadabra", "bra", "", true },
+                { "(?xi)[hello world()]", "hello", null, true }
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("invokeInvalidRegexTestData")
+    void invokeInvalidRegexTest(String input, String pattern, String flags, Class<?> expectedErrorEventClass) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: Inline parameter expectedErrorEventClass. it is always going to be InvalidParametersEvent.class.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java [hunk 8]
```diff
@@ -19,116 +19,146 @@
...
+                { "testString", "(?=\\\\s)", null, InvalidParametersEvent.class },
+                { "fo\nbar", "(?iU)(?iU)(ab)[|cd]", null, InvalidParametersEvent.class },
+                { "fo\nbar", "(?x)(?i)hello world", "i", InvalidParametersEvent.class },
+                { "fo\nbar", "(?xi)hello world", null, InvalidParametersEvent.class },
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("invokeWithoutFlagsNotMatchTestData")
+    void invokeWithoutFlagsNotMatchTest(String input, String pattern, String flags, Boolean expectedResult) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java:110]: Inline parameter expectedResult. it is always going to be false.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java [hunk 9]
```diff
@@ -19,116 +19,146 @@
...
+                { "foobar", "^fo*bb", null, false },
+                { "h", "(.)\3", null, false },
+                { "h", "(.)\2", null, false },
+                { "input", "\3", null, false }
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("invokeWithFlagCaseInsensitiveTestData")
+    void invokeWithoutFlagsPatternTest(String input, String pattern, String flags, Boolean expectedResult) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java:126]: Inline parameter expectedResult. it is always going to be true.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java [hunk 10]
```diff
@@ -19,62 +19,88 @@
...
+                { "testString", "test", null, "s", InvalidParametersEvent.class },
+                { null, "test", null, "s", InvalidParametersEvent.class },
+                { null, "test", "ttt", "s", InvalidParametersEvent.class },
+                { null, null, "ttt", "s", InvalidParametersEvent.class },
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("invokeUnsupportedFlagsTestData")
+    void invokeUnsupportedFlagsTest(String input, String pattern, String replacement, String flags, Class<?> expectedErrorEventClass) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java]: Inline parameter expectedErrorEventClass. it is always going to be InvalidParametersEvent.class.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java [hunk 11]
```diff
@@ -19,62 +19,88 @@
...
+        };
     }
 
-    @Test
-    void invokeWithoutFlagsPatternMatches() {
-        FunctionTestUtil.assertResult(replaceFunction.invoke("testString", "^test", "ttt"), "tttString");
-        FunctionTestUtil.assertResult(replaceFunction.invoke("testStringtest", "^test", "ttt"), "tttStringtest");
+    @ParameterizedTest
+    @MethodSource("invokeNullWithFlagsTestData")
+    void invokeNullWithFlagsTest(String input, String pattern, String replacement, String flags, Class<?> expectedErrorEventClass) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java]: Inline parameter expectedErrorEventClass. it is always going to be InvalidParametersEvent.class.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java [hunk 12]
```diff
@@ -19,62 +19,88 @@
...
-        FunctionTestUtil.assertResultError(replaceFunction.invoke("testString", null, null),
-                                           InvalidParametersEvent.class);
-        FunctionTestUtil.assertResultError(replaceFunction.invoke("testString", "test", null),
-                                           InvalidParametersEvent.class);
-        FunctionTestUtil.assertResultError(replaceFunction.invoke(null, "test", null), InvalidParametersEvent.class);
-        FunctionTestUtil.assertResultError(replaceFunction.invoke(null, "test", "ttt"), InvalidParametersEvent.class);
-        FunctionTestUtil.assertResultError(replaceFunction.invoke(null, null, "ttt"), InvalidParametersEvent.class);
+    @ParameterizedTest
+    @MethodSource("invokeNullTestData")
+    void invokeNullTest(String input, String pattern, String replacement, Class<?> expectedErrorEventClass) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java]: Inline parameter expectedErrorEventClass. it is always going to be InvalidParametersEvent.class.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+        return new Object[][] {
+                { "matches('test', '^test', 'i')", Boolean.class, true },
+                { "matches('fo\\nbar', 'o.b', '')", Boolean.class, false },
+                { "replace('testString', '^test', 'ttt', '')", String.class, "tttString" }
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("evaluateXQueryExpressionInvalidParametersTestData")
+    void evaluateXQueryExpressionInvalidParametersTest(String expression, Class<?> returnTypeClass, Class<?> expectedException) {
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java]: Inline parameter expectedErrorEventClass. it is always going to be UnsupportedOperationException.class.

**Review by pibizza — CHANGES_REQUESTED** [review]
Mostly some fixes to make the tests clearer and crisper. 
- very good to use parametric tests
- do not use parameters that are constants - if they do not change they should not be parameters
- consider adding a parametric name for the tests, this would make it easier to read them. See https://junit.org/junit5/docs/current/user-guide/#writing-tests-display-names for reference.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java (hunk 3 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: @pibizza Thank you for the review! I'm aware that all the cases you reported are managing constants, but I did that to ease a future extension of those tests. It's just a case some of them are currently constant values. Does make sense?

Comment (pibizza) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: @yesamer Yes, that makes sense. Feel free to keep the code as it is in that case.

**Comment by yesamer** [comment]
@pibizza I applied your suggestion in all the places where I'm sure the parameter will always be a constant (basically all the exceptions). Thank you!

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java (hunk 3 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: Done

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java (hunk 4 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: Done

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java (hunk 7 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/MatchesFunctionTest.java]: Done

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java (hunk 10 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java]: Done

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java (hunk 11 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java]: Done

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java (hunk 12 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/functions/ReplaceFunctionTest.java]: Done

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java (hunk 13 above)
Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/util/XQueryImplUtilTest.java]: Done

**Review by pibizza — APPROVED** [review]
LGTM

**Review by mariofusco — APPROVED** [review]

**Comment by yesamer** [comment]
Failures: 
```
apache_incubator-kie-kogito-runtimes/quarkus/integration-tests/integration-tests-quarkus-processes/src/test/java/org/kie/kogito/integrationtests/quarkus/ProcessSvgAddonIT.java
apache_incubator-kie-kogito-runtimes/springboot/addons/process-svg/src/test/java/org/kie/kogito/svg/service/SpringBootProcessSvgServiceTest
```

 on runtimes are unrelated to these PR changes.

**Pull request merged**